File: java/socketd/src/main/java/org/noear/socketd/core/impl/MessageDefault.java
package org.noear.socketd.core.impl;

import org.noear.socketd.core.Constants;
import org.noear.socketd.core.Entity;
import org.noear.socketd.core.Flag;
import org.noear.socketd.core.Message;


/**
 * 消息默认实现（帧[消息[实体]]）
 *
 * @author noear
 * @since 2.0
 */
public class MessageDefault implements Message {
    private String key = Constants.DEF_KEY;
    private String topic = Constants.DEF_TOPIC;
    private Entity entity = null;

    private Flag flag = Flag.Unknown;

    public Flag getFlag() {
        return flag;
    }

    public MessageDefault flag(Flag flag) {
        this.flag = flag;
        return this;
    }

    public MessageDefault key(String key) {
        this.key = key;
        return this;
    }

    public MessageDefault topic(String topic) {
        this.topic = topic;
        return this;
    }

    public MessageDefault entity(Entity entity) {
        this.entity = entity;
        return this;
    }


    /**
     * 是否为请求
     */
    public boolean isRequest() {
        return flag == Flag.Request;
    }

    /**
     * 是否为订阅
     */
    public boolean isSubscribe() {
        return flag == Flag.Subscribe;
    }

    /**
     * Key
     */
    public String getKey() {
        return key;
    }

    /**
     * topic
     */
    public String getTopic() {
        return topic;
    }

    /**
     * entity
     */
    @Override
    public Entity getEntity() {
        return entity;
    }

    @Override
    public String toString() {
        return "Message{" +
                "key='" + key + '\'' +
                ", topic='" + topic + '\'' +
                ", entity=" + entity +
                '}';
    }
}

File: python/socketd/socketd/core/module/MessageDefault.py
from io import BytesIO
from typing import Optional, Any, Dict

from .Entity import Entity
from .Message import Message
from ..Costants import Constants, Flag


class MessageDefault(Message):
    def __init__(self):
        self.sid = Constants.DEF_SID
        self.event = Constants.DEF_EVENT
        self.entity: Entity = None
        self.flag = Flag.Unknown

    def get_flag(self):
        return self.flag

    def set_flag(self, flag):
        self.flag = flag
        return self

    def set_sid(self, sid):
        self.sid = sid
        return self

    def set_event(self, event):
        self.event = event
        return self

    def set_entity(self, entity):
        self.entity = entity
        return self

    def is_request(self):
        return self.flag == Flag.Request

    def is_subscribe(self):
        return self.flag == Flag.Subscribe

    def is_close(self):
        return self.flag == Flag.Close

    def get_sid(self):
        return self.sid

    def get_event(self):
        return self.event

    def get_entity(self):
        return self.entity

    def __str__(self):
        return f"Message{{sid='{self.sid}', event='{self.event}', entity={self.entity}}}"

    def get_meta_string(self) -> str:
        return self.entity.get_meta_string()

    def get_meta_map(self) -> Dict[str, str]:
        return self.entity.get_meta_map()

    def get_meta(self, name: str) -> Optional[Any]:
        return self.entity.get_meta(name)

    def get_meta_or_default(self, name: str, default: str) -> str:
        return self.entity.get_meta_or_default(name, default)

    def get_data(self) -> BytesIO:
        return self.entity.get_data()

    def get_data_as_string(self) -> str:
        return self.entity.get_data_as_string()

    def get_data_size(self) -> int:
        return self.entity.get_data_size()

    def get_data_as_bytes(self) -> bytes:
        return self.entity.get_data_as_bytes()